Nakama/Client.cs

```diff
@@ -594,217 +690,360 @@ public async Task LinkFacebookAsync(ISession session, string token, bool? import
+                {
+                    r.Username = session.Username;
+                }
+            }
+
+            return response;
         }
 
         /// <inheritdoc cref="ListMatchesAsync"/>
         public async Task<IApiMatchList> ListMatchesAsync(ISession session, int min, int max, int limit,
-            bool authoritative, string label, string query, RetryConfiguration retryConfiguration = null, CancellationToken canceller = default)
+            bool authoritative, string label, string query, RetryConfiguration retryConfiguration = null,
+            CancellationToken canceller = default)
         {
             if (AutoRefreshSession && !string.IsNullOrEmpty(session.RefreshToken) &&
                 session.HasExpired(DateTime.UtcNow.Add(DefaultExpiredTimeSpan)))
             {
                 await SessionRefreshAsync(session, null, retryConfiguration, canceller);
             }
 
-            return await  _retryInvoker.InvokeWithRetry(() => _apiClient.ListMatchesAsync(session.AuthToken, limit, authoritative, label, min, max, query, canceller), new RetryHistory(session, retryConfiguration ?? GlobalRetryConfiguration, canceller));
+            return await _retryInvoker.InvokeWithRetry(
+                () => _apiClient.ListMatchesAsync(session.AuthToken, limit, authoritative, label, min, max, query,
+                    canceller), new RetryHistory(session, retryConfiguration ?? GlobalRetryConfiguration, canceller));
         }
 
         /// <inheritdoc cref="ListNotificationsAsync"/>
         public async Task<IApiNotificationList> ListNotificationsAsync(ISession session, int limit = 1,
-            string cacheableCursor = null, RetryConfiguration retryConfiguration = null, CancellationToken canceller = default)
+            string cacheableCursor = null, RetryConfiguration retryConfiguration = null,
+            CancellationToken canceller = default)
         {
             if (AutoRefreshSession && !string.IsNullOrEmpty(session.RefreshToken) &&
                 session.HasExpired(DateTime.UtcNow.Add(DefaultExpiredTimeSpan)))
             {
                 await SessionRefreshAsync(session, null, retryConfiguration, canceller);
             }
 
-            return await  _retryInvoker.InvokeWithRetry(() => _apiClient.ListNotificationsAsync(session.AuthToken, limit, cacheableCursor, canceller), new RetryHistory(session, retryConfiguration ?? GlobalRetryConfiguration, canceller));
+            var response = await _retryInvoker.InvokeWithRetry(
+                () => _apiClient.ListNotificationsAsync(session.AuthToken, limit, cacheableCursor, canceller),
+                new RetryHistory(session, retryConfiguration ?? GlobalRetryConfiguration, canceller));
+
+            foreach (var notification in response.Notifications)
+            {
+                if (!session.Username.Equals(notification.SenderId) && notification is ApiNotification n)
+                {
+                    n.Subject = session.UserId;
```

Comment:
Isn't subject supposed to be an arbitrary string, i.e. "You earned X coins!", no particular relationship to the user information?